src/Units/UnitModel.java

```diff
@@ -21,45 +20,39 @@
   private static final long serialVersionUID = 1L;
 
   /**
-   * Defines the core types of units we expect to be in any given unit set.
-   * Unit sets should have no less than one of each.
-   * Unit sets should order models such that the most "sensible" type (generally cheapest, land) is the first type of a given role.
+   * These are the high-level traits on which to filter unit types.
+   * Unit sets should order models such that the most "sensible" type
+   *   (generally cheapest, land, assault) is the first type fulfill a given role.
    */
-  public enum UnitRoleEnum
-  {
-    INFANTRY, MECH,
-    RECON, ASSAULT,
-    SIEGE, ANTI_AIR,
-    AIR_ASSAULT, AIR_SUPERIORITY,
-    TRANSPORT,
-  };
-
-  private static ArrayList<String> unitRoleIDs = null;
-  public static ArrayList<String> getUnitRoleIDs()
-  {
-    if( null == unitRoleIDs )
-    {
-      unitRoleIDs = new ArrayList<String>();
-      for( UnitRoleEnum role : UnitRoleEnum.values() )
-        unitRoleIDs.add(standardizeID(role.toString()));
-    }
-    return unitRoleIDs;
-  }
+  // Morphology
+  public static final long TROOP           = 1 <<  0; // Not in vehicle
+  public static final long TANK            = 1 <<  1; // Ground-traveling vehicle
+  public static final long HOVER           = 1 <<  2; // e.g. helicopter
+  public static final long JET             = 1 <<  3;
+  public static final long SHIP            = 1 <<  4;
+  // Domain
+  public static final long LAND            = 1 <<  8;
+  public static final long AIR_LOW         = 1 <<  9;
+  public static final long AIR_HIGH        = 1 << 10;
+  public static final long SEA             = 1 << 11;
+  public static final long SUBSURFACE      = 1 << 12;
+  // Role; Unit sets are expected to have at least one of each
+  public static final long MECH            = 1 << 16; // Footsoldier equipped against hardened targets
+  public static final long RECON           = 1 << 17; // Scout
+  public static final long ASSAULT         = 1 << 18; // Fast unit that can deal with hardened ground targets
+  public static final long SIEGE           = 1 << 19; // Unit with heavy strike capabilities at range
```

Comment:
So Snipers would not count as `SIEGE`?

Reply:
Nah, I don't think so. Meridian doesn't want tanks transforming into snipers, anyhow.
I'm... not sure snipers really have enough range to qualify for 2 range, and they aren't proper indirect-fire, so I dunno that I'd want to bring them in to a unit set with similar assumptions as the existing ones.

Reply:
Yeah, well I'm also not super-comfortable with having Meridian's ability key off of unit role alone - seems fragile if we ever do add new unit options. Can she not choose those units by name?

Reply:
I would argue that basing it off of unit name makes it *more* fragile.
If I, say, wanted to steal Wargroove's units, the existing implementation would match up Archers and Knights.

Also, my comment above this block does address the fragility somewhat - Unit models should be ordered such that fetching the first unit of a given role is a sensical operation in most cases.

Reply:
Well, fair enough. Different approaches will leave us vulnerable to different kinds of change. I guess I won't quibble too much over this until we have a real use case that forces our hand.